docs/source/miscellaneous/stubs.rst

```diff
@@ -0,0 +1,610 @@
+#######
+ Stubs
+#######
+
+Stub files are pieces of plain PHP code which only contain declarations without actually runnable
+code. A very basic stub looks like this:
+
+.. code:: php
+
+   /** @var string */
+   const FOO = "foo";
+   /** @var int */
+   const BAR = "bar";
+
+   class Foo {
+       public function bar(): string {}
+   }
+
+   function foo(string $param): string {}
+
+Any kind of symbols can be declared via stubs, and with the exception of disjunctive normal form
+(DNF) types, basically any kind of type declaration is supported. Additional meta information can be
+added either via PHPDoc or attributes. Namespaces can also be used by either adding a top-level
+``namespace`` declaration or by using namespace blocks:
+
+.. code:: php
+
+   namespace {
+       /** @var string */
+       const FOO = "foo";
+       /** @var string */
+       const BAR = "bar";
+
+       class Foo {
+           public function bar(): string {}
+       }
+   }
+
+   namespace Foo {
+       function foo(string $param): string {}
+   }
+
+The above example declares the global constants and class ``Foo`` in the top-level namespace, while
+``foo()`` will be available in the ``Foo`` namespace.
+
+********************
+ Using gen_stub.php
+********************
+
+By convention, stub files have a ``.stub.php`` extension. They are processed by
+``build/gen_stub.php``: it uses PHP-Parser_ for parsing, then depending on the configuration and the
```

Comment:
```suggestion
``build/gen_stub.php``: it uses PHP-Parser for parsing, then depending on the configuration and the
```
Seems to be some weird issue here?

Reply:
Hmm, it is supposed to be an external link: https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html but I accidentally forgot about the ` pair

Reply:
Looks like rstfmt removed the ticks. I guess they are unnecessary